### ISSUE TITLE
Dovecot 2.3+ needs SSL chain in cert instead of ssl_ca

### ISSUE THREAD

**Issue by infernix** [virtualmin/virtualmin-gpl#157]
In local_name config here:

https://github.com/virtualmin/virtualmin-gpl/blob/73c52654184bbb50eaa4b78d2a18fc82639e7c23/feature-ssl.pl#L1884

There should be a check which version of Dovecot is in place. Starting in 2.3, ssl_ca is not to be used for the certificate chain. Instead, the chain should be part of the ssl_cert file.

The easiest fix is to point it to ssl.combined instead of ssl.cert if dovecot >= 2.3, so should be fairly simple to resolve.

**Comment by jcameron** [comment]
Thanks, we'll take a look ..

**Comment by jcameron** [comment]
Can you point me to some documentation on this change to the ssl_ca directive? I wasn't able to find a mention of it one way or another in the Dovecot docs.

**Comment by infernix** [comment]
Unfortunately no, I discovered it myself during Debian 9 > 10 upgrade where it broke, and ended up on these threads:

https://dovecot.org/pipermail/dovecot/2018-September/112881.html (by dovecot author)

https://dovecot.org/pipermail/dovecot/2019-August/116719.html (also by dovecot author)

It kind of implies it should have also been that way on 2.2 and below but I can't confirm that as I haven't tested it.

**Comment by iliarostovtsev** [comment]
The odd thing, as I'm running **Dovecot 2.3.10** on my Fedora Server, where there is no any issues with using **ssl_ca** directive and it works just fine. Besides, there is nothing mentioned on [upgrade guide](https://wiki2.dovecot.org/Upgrading/2.3) from 2.2 to 2.3.

There must have been some mistake.

**Comment by infernix** [comment]
Maybe there's something different with the Debian package vs the Fedora one, but the mailinglist responses are by the guy who wrote the code.

It's also kind of indicated in the config thoug, see https://github.com/dovecot/core/blob/2faf3f4e05bf081fcf72f02fa2cd0c8979627c0c/doc/example-config/conf.d/10-ssl.conf#L21:

```
# PEM encoded trusted certificate authority. Set this only if you intend to use
# ssl_verify_client_cert=yes. The file should contain the CA certificate(s)
# followed by the matching CRL(s). (e.g. ssl_ca = </etc/ssl/certs/ca.pem)
#ssl_ca = 
```

This says `ssl_ca` is to be used if you are doing *client* ssl verification. So shouldn't have anything to do with the SSL cert served by the daemon(s).

**Comment by jcameron** [comment]
Yeah it's odd that even older documentation doesn't mention that `ssl_ca` option inside a per-IP block, but it does work.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
